pkg/bootstrap/config.go

```diff
@@ -306,9 +306,14 @@ func getProxyConfigOptions(config *meshAPI.ProxyConfig, metadata *model.Bootstra
 			opts = append(opts, option.DataDogAddress(tracer.Datadog.Address))
 		case *meshAPI.Tracing_Stackdriver_:
 			var projectID string
-			var err error
-			if projectID, err = md.ProjectID(); err != nil {
-				return nil, fmt.Errorf("unable to process Stackdriver tracer: %v", err)
+			var projFound bool
+
+			if projectID, projFound = metadata.PlatformMetadata[platform.GCPProject]; !projFound {
+				projectID, projFound = metadata.PlatformMetadata[platform.GCPProjectNumber]
+			}
+
+			if !projFound {
+				return nil, errors.New("unable to process Stackdriver tracer: missing GCP Project")
```

Comment:
This breaks in non GCE environment (and the integ test stackdriver/vm test) since if !projectFound is always true in non GCE environment.
We need to check if the environment is on GCE, and return error only if it is running on GCE environment.

Reply:
We need to identify why, in your test runs, the fake metadata server is not working.  `!projFound` _should not_ always be true, with the setup to supply a fake metadata server for those envs.